File: code_challenge_solution/c++/144 Password & passphrase generator.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

string generatePassword(int length) {
    string password = "";
    string characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()";

    srand(time(0));

    for (int i = 0; i < length; i++) {
        int randomIndex = rand() % characters.length();
        password += characters[randomIndex];
    }

    return password;
}

string generatePassphrase(int length) {
    string passphrase = "";
    string words[] = {"apple", "banana", "orange", "grape", "watermelon", "kiwi", "pineapple", "mango", "strawberry", "blueberry"};

    srand(time(0));

    for (int i = 0; i < length; i++) {
        int randomIndex = rand() % 10;
        passphrase += words[randomIndex];

        if (i < length - 1) {
            passphrase += " ";
        }
    }

    return passphrase;
}

int main() {
    int choice;
    int length;

    cout << "Password & Passphrase Generator" << endl;
    cout << "1. Generate Password" << endl;
    cout << "2. Generate Passphrase" << endl;
    cout << "Enter your choice: ";
    cin >> choice;

    if (choice == 1) {
        cout << "Enter the length of the password: ";
        cin >> length;
        cout << "Generated Password: " << generatePassword(length) << endl;
    } else if (choice == 2) {
        cout << "Enter the length of the passphrase: ";
        cin >> length;
        cout << "Generated Passphrase: " << generatePassphrase(length) << endl;
    } else {
        cout << "Invalid choice" << endl;
    }

    return 0;
}

File: code_challenge_solution/python/144 Password & passphrase generator.py
import random
import string

def generate_password(length):
    characters = string.ascii_letters + string.digits + string.punctuation
    password = ''.join(random.choice(characters) for _ in range(length))
    return password

def generate_passphrase(num_words):
    with open('wordlist.txt', 'r') as wordlist_file:
        wordlist = wordlist_file.readlines()

    passphrase = ''
    for _ in range(num_words):
        word = random.choice(wordlist).strip()
        passphrase += word + ' '

    return passphrase.strip()

print("Generated Password:", generate_password(10))
print("Generated Passphrase:", generate_passphrase(5))